### PR TITLE
Update CHANGELOG.md

### LINKED ISSUE #3199: Review change log for 4.1.0
Gentlemen, I'm planning to do the next release in the immediate future.

One thing I'd like to do differently for this release is to review the change log _before_ we do the release. Previously, we updated the change log as part of the release project, which made review of the change log itself impossible, because by the time the changes were visible, the new version was already on PyPI and it was too late to make changes.

This time around, I've kept the change log up to date as we merged each PR. This has two benefits:

1. Anyone can spot errors in the change log and correct them.
2. Anyone can see how many of each issue type (bug fix, documentation, improvement, etc) we've made since the last release. Previously, this information was only visible via the issue tracker, and only a select few bother to dig around there.

Can you please have a look at the change log and make any required changes? Things like wording, omission of less relevant PRs, wrong PR category, missed PR, etc.

For the last part, here is the list of merged PRs since the last release: https://github.com/RaRe-Technologies/gensim/pulls?q=is%3Apr+is%3Aclosed+closed%3A%3E2021-04-01

### PR DESCRIPTION
**Pull request by mpenkov**
Fix #3199

### CODE REVIEW

**Comment by mpenkov** [comment]
@piskvorky Please have a look. I think this is the last thing left before we do the actual release.

File: CHANGELOG.md [hunk 1]
```diff
@@ -108,7 +108,7 @@ Plus a large number of smaller improvements and fixes, as usual.
 
 * [#3176](https://github.com/RaRe-Technologies/gensim/pull/3176): Eliminate obsolete step parameter from doc2vec infer_vector and similarity_unseen_docs, by [@rock420](https://github.com/rock420)
 * [#2965](https://github.com/RaRe-Technologies/gensim/pull/2965): Remove strip_punctuation2 alias of strip_punctuation, by [@sciatro](https://github.com/sciatro)
-
+* [#3180](https://github.com/RaRe-Technologies/gensim/pull/3180): Move  preprocessing functions from gensim.corpora.textcorpus and gensim.corpora.lowcorpus to gensim.parsing.preprocessing, by [@rock420](https://github.com/rock420)
```
Comment (piskvorky) [inline CHANGELOG.md:111]:
```suggestion
* [#3180](https://github.com/RaRe-Technologies/gensim/pull/3180): Move preprocessing functions from gensim.corpora.textcorpus and gensim.corpora.lowcorpus to gensim.parsing.preprocessing, by [@rock420](https://github.com/rock420)
```

**Review by piskvorky — APPROVED** [review]

**Comment by mpenkov** [comment]
Yeah, need to get aarm wheels building.

I've been busy, will try to look into it next week.

**Comment by mpenkov** [comment]
There's one build failing, looking into it now: https://app.travis-ci.com/github/RaRe-Technologies/gensim/jobs/532471077

After that's fixed we can assemble the release, push it to PyPI and then it's 🍾 and 🚬

**Pull request merged**